packages/amplify-category-function/src/provider-utils/awscloudformation/utils/updateDependentFunctionCfn.ts

```diff
@@ -0,0 +1,85 @@
+import { $TSContext, $TSObject, JSONUtilities } from 'amplify-cli-core';
+import { FunctionParameters } from 'amplify-function-plugin-interface';
+import { getResourcesForCfn, generateEnvVariablesForCfn } from '../service-walkthroughs/execPermissionsWalkthrough';
+import { updateCFNFileForResourcePermissions } from '../service-walkthroughs/lambda-walkthrough';
+import { loadFunctionParameters } from './loadFunctionParameters';
+import path from 'path';
+import { functionParametersFileName } from './constants';
+import { category } from '../../../constants';
+
+export async function updateDependentFunctionsCfn(
```

Comment:
would recommend breaking this function into a few helper functions. 3 nested loops is a lot to follow